io/src/main/java/org/commonjava/maven/ext/io/PomIO.java

```diff
@@ -73,8 +79,23 @@
 
     private final JDOMModelConverter jdomModelConverter = new JDOMModelConverter( );
 
+    private final boolean parsePomTemplates;
+
     private String manifestComment;
 
+    @Inject
+    public PomIO(MavenSessionHandler handler )
+    {
+        parsePomTemplates = Boolean.parseBoolean(
+                        handler.getUserProperties().getProperty( PARSE_POM_TEMPLATES, "true" ) );
+    }
+
+    // Test use only.
+    public PomIO()
```

Comment:
I am not sure I see this constructor used anywhere currently. If it's used within the same package, it could be made package-private.

Reply:
Vast amounts of tests use it so seemed simpler to create extra ctor.

Reply:
Oh, they were using the default constructor? I think the some version of Java is getting rid of those and they have to be written explicitly.

Reply:
But for now I think its acceptable?

Reply:
Yes, it was just for your information, and we have an explicit constructor now.

I don't know if it will ever be removed, but there are some bugs that allow a warning to be enabled as of Java 16, and specifically for public APIs.

https://bugs.openjdk.java.net/browse/JDK-8071961
https://bugs.openjdk.java.net/browse/JDK-8250795
https://bugs.openjdk.java.net/browse/JDK-8252348

Reply:
Thanks.